src/benchmarks/micro/runtime/StoreBlock/StoreBlock.cs

```diff
@@ -0,0 +1,1244 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+[BenchmarkCategory(Categories.Runtime, Categories.JIT)]
+public class StoreBlock
+{
+    const int OperationsPerInvoke = 1000;
+
+    byte[] _srcData = new byte[1024];
+    byte[] _dstData = new byte[1024];
+
+    [GlobalSetup]
+    public void Setup()
+    {
+        for (int i = 0; i < _srcData.Length; i++)
+            _srcData[i] = (byte)(i % 256);
+
+        for (int i = 0; i < _dstData.Length; i++)
+            _dstData[i] = (byte)(i % 256);
+    }
+
+    [StructLayout(LayoutKind.Explicit, Size=8)]
+    struct Struct8
+    {
+    }
+
+    Struct8 fld8;
+
+    [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
+    public void InitBlockAllZerosHeap8()
+    {
+        for (int i = 0; i < OperationsPerInvoke; i++)
+        {
+            Unsafe.InitBlock(ref _dstData[0], 0, 8);
+        }
+    }
+
+    [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
+    public unsafe void InitBlockAllZerosLocalAddr8()
+    {
+        Struct8 dstLcl;
+
+        for (int i = 0; i < OperationsPerInvoke; i++)
+        {
+            Unsafe.InitBlock(&dstLcl, 0, 8);
+        }
+
+        fld8 = dstLcl;
+    }
+
+    [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
+    public void InitBlockAllOnesHeap8()
+    {
+        for (int i = 0; i < OperationsPerInvoke; i++)
+        {
+            Unsafe.InitBlock(ref _dstData[0], 255, 8);
+        }
+    }
+
+    [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
+    public unsafe void InitBlockAllOnesLocalAddr8()
+    {
+        Struct8 dstLcl;
+
+        for (int i = 0; i < OperationsPerInvoke; i++)
+        {
+            Unsafe.InitBlock(&dstLcl, 1, 8);
+        }
+
+        fld8 = dstLcl;
+    }
+
+    [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
+    public void CopyBlockHeap8()
+    {
+        for (int i = 0; i < OperationsPerInvoke; i++)
+        {
+            Unsafe.CopyBlock(ref _dstData[0], ref _srcData[0], 8);
+        }
+    }
+
+    [Benchmark(OperationsPerInvoke = OperationsPerInvoke)]
+    public unsafe void CopyBlockLocalAddr8()
```

Comment:
Are we executing a different `Unsafe.CopyBlock` code path compared to the `CopyBlockHeap8` above?